mapie/estimators.py

```diff
@@ -565,53 +574,57 @@ def predict(self, X: ArrayLike) -> np.ndarray:
             ]
         )
         X = check_array(X, force_all_finite=False, dtype=["float64", "object"])
-        alpha = self._check_alpha(self.alpha)
-
-        # SHAPES:
-        # (n_samples_train,) : self.residuals_
-        # (n_alpha,) : alpha
-        # (n_samples_test, n_alpha) : y_pred_low, y_pred_up
-        # (n_samples_test, n_samples_train) : y_pred_multi, lower/upper_bounds
-        n_alpha = len(alpha)
         y_pred = self.single_estimator_.predict(X)
-        # At this point, y_pred is of shape (n_samples_test,)
-        if self.method in ["naive", "base"] or self.cv == "prefit":
-            quantile = np.quantile(
-                self.residuals_, 1 - alpha, interpolation="higher"
-            )
-            y_pred_low = y_pred[:, np.newaxis] - quantile
-            y_pred_up = y_pred[:, np.newaxis] + quantile
+
+        if alpha is None:
+            return np.array(y_pred)
         else:
-            y_pred_multi = np.column_stack(
-                [e.predict(X) for e in self.estimators_]
-            )
-            if self.method == "plus":
-                # At this point,
-                # y_pred_multi is of shape (n_samples_test, n_estimators_)
-                # We thus enforce y_pred_multi
-                # to be of shape (n_samples_test, n_samples_train)
-                if len(self.estimators_) < len(self.k_):
-                    y_pred_multi = y_pred_multi[:, self.k_]
-                lower_bounds = y_pred_multi - self.residuals_
-                upper_bounds = y_pred_multi + self.residuals_
-            if self.method == "minmax":
-                lower_bounds = np.min(y_pred_multi, axis=1, keepdims=True)
-                upper_bounds = np.max(y_pred_multi, axis=1, keepdims=True)
-                lower_bounds = lower_bounds - self.residuals_
-                upper_bounds = upper_bounds + self.residuals_
-            y_pred_low = np.column_stack([
-                np.quantile(
-                    lower_bounds, _alpha, axis=1, interpolation="lower"
-                ) for _alpha in alpha
-            ])
-            y_pred_up = np.column_stack([
-                np.quantile(
-                    upper_bounds, 1 - _alpha, axis=1, interpolation="higher"
-                ) for _alpha in alpha
-            ])
-            if self.ensemble:
-                y_pred = np.median(y_pred_multi, axis=1)
-        y_pred = np.column_stack([y_pred]*n_alpha)
-        # At this point, y_pred is of shape (n_samples_test, n_alpha)
-        y_preds = np.stack([y_pred, y_pred_low, y_pred_up], axis=1)
-        return y_preds
+            # SHAPES:
+            # (n_samples_train,) : self.residuals_
+            # (n_alpha,) : alpha
+            # (n_samples_test, n_alpha) : y_pred_low, y_pred_up
+            # (n_samples_test, n_samples_train) : y_pred_multi, low/up_bounds
+            alpha_np = self._check_alpha(alpha)
```

Comment:
Pour le coup, je laisserais le `_check_alpha` là où il était (en deuxième ligne), en laissant la possibilité de mettre des `None`. Peut-on également faire un remplacement global de `alpha_np` par `alpha_` ? Ce serait plus concis

Reply:
Modifié.

Reply:
Quel est l'intéret de faire un `check_alpha` si alpha est `None` ? Cela complique inutilement la fonction `_check_alpha`.

Reply:
C'est pour être homogène avec `_check_estimator` dans l'usage. Le check doit toujours être la première ligne de code de la fonction.

Reply:
Dans ce cas, je pense que nous sommes obligés de retour un alpha de type `np.ndarray` même si `alpha is None` car sinon mypi râle lors du calcul de `y_pred_low` et `y_pred_up`.